lib/dal/src/prop.rs

```diff
@@ -926,4 +941,99 @@ impl Prop {
         }
         Ok(prop)
     }
+
+    pub async fn direct_child_props_ordered(
+        ctx: &DalContext,
+        prop_id: PropId,
+    ) -> PropResult<Vec<Prop>> {
+        let workspace_snapshot = ctx.workspace_snapshot()?;
+
+        let kind = workspace_snapshot
+            .get_node_weight_by_id(prop_id)
+            .await?
+            .get_prop_node_weight()?
+            .kind();
+
+        let ordered_child_props = match kind {
+            PropKind::Boolean | PropKind::Integer | PropKind::String => Vec::new(),
+            PropKind::Array | PropKind::Map | PropKind::Object => {
+                let ordered_child_prop_ids = workspace_snapshot
+                    .ordered_children_for_node(prop_id)
+                    .await?
+                    .ok_or(WorkspaceSnapshotError::OrderingNotFound(prop_id.into()))?;
+
+                let mut ordered_child_props = Vec::new();
+                for ordered_child_prop_id in ordered_child_prop_ids {
+                    if let NodeWeight::Prop(child_prop_weight) = workspace_snapshot
+                        .get_node_weight_by_id(ordered_child_prop_id)
+                        .await?
+                    {
+                        let child_prop =
+                            Self::get_by_id_or_error(ctx, child_prop_weight.id().into()).await?;
+                        ordered_child_props.push(child_prop);
+                    }
+                }
+                ordered_child_props
+            }
+        };
```

Comment:
Collect child props in order. This contains safety rails in case you accidentally call this from a `Prop` whose type is a scalar.